pgjdbc/src/main/java/org/postgresql/util/PGbytea.java

```diff
@@ -184,6 +184,9 @@ public static String toPGLiteral(Object value) throws IOException {
         appendHexString(sb, bytes, sw.getOffset(), length);
       } else if (length > 0) {
         InputStream str = sw.getStream();
+        if ( str.markSupported() ) {
+          str.mark(Integer.MAX_VALUE);
+        }
```

Comment:
The stream might not support mark/reset, so this workaround does not look reliable.

Reply:
So the alternative would be to read the stream into a buffer. I'm not a fan of that. I'm sort of willing to leave this as a half-baked work around.
Do you have any other suggestions?

Reply:
`mark(max_value)` in most of the cases buffers the data in memory :-/

Reply:
works well for `ByteArrayInputStream` what I am worried about is a FileInputStream that could potentially blow up the client.